BasiliskII/src/Unix/ether_unix.cpp

```diff
@@ -334,12 +330,20 @@ bool ether_init(void)
 			WarningAlert(str);
 			goto open_error;
 		}
-		net_if_name = strdup(ifr.ifr_name);
+		net_if_name = strdup(name);
 		if (!execute_network_script("up")) {
 			sprintf(str, GetString(STR_TUN_TAP_CONFIG_WARN), "script execute error");
 			WarningAlert(str);
 			goto open_error;
 		}
+		memset(&ifr, 0, sizeof(ifr));
+		ifr.ifr_flags = IFF_TAP | IFF_NO_PI;
+		strcpy(ifr.ifr_name, name);
```

Comment:
What ensures that ifr_name is big enough to hold name? Please use strlcpy instead.